src/dfx/src/lib/extension/manager/install.rs

```diff
@@ -0,0 +1,99 @@
+use crate::lib::error::extension::ExtensionError;
+use crate::lib::extension::{manager::ExtensionManager, manifest::ExtensionCompatibilityMatrix};
+
+use flate2::read::GzDecoder;
+use reqwest::Url;
+use semver::{BuildMetadata, Prerelease, Version};
+use tar::Archive;
+use tempfile::{tempdir_in, TempDir};
+
+use std::io::Cursor;
+#[cfg(not(target_os = "windows"))]
+use std::os::unix::fs::PermissionsExt;
+
+impl ExtensionManager {
+    pub fn install_extension(&self, extension_name: &str) -> Result<(), ExtensionError> {
+        if self.get_extension_directory(extension_name).exists() {
+            return Err(ExtensionError::ExtensionAlreadyInstalled(format!(
+                "extension {} already installed",
+                extension_name
+            )));
```

Comment:
Move `extension {} already installed` text to the `#[error()]` directive.
```suggestion
            return Err(ExtensionError::ExtensionAlreadyInstalled(
                extension_name
            ));
```